src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirDataStore.cs

```diff
@@ -129,14 +129,19 @@ internal class SqlServerFhirDataStore : IFhirDataStore, IProvideCapability
                 resource.SearchIndices?.ToLookup(e => _searchParameterTypeMap.GetSearchValueType(e)),
                 resource.LastModifiedClaims);
 
+            const int deadlockVictim = 1205;
+            const int maxRetryCount = 5;
+            int retryCount = 0;
+            int baseDelay = 5;
+
             while (true)
             {
                 cancellationToken.ThrowIfCancellationRequested();
 
                 // ** We must use CreateNonRetrySqlCommand here because the retry will not reset the Stream containing the RawResource, resulting in a failure to save the data
                 using (SqlConnectionWrapper sqlConnectionWrapper = await _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken, true))
                 using (SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateNonRetrySqlCommand())
```

Comment:
>using (SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateNonRetrySqlCommand()) [](http://example.com/codeflow?start=16&length=93)

Has approach with using retryable SQL command and raw resource as binary array considered?

Reply:
That would definitely be much more work than I thought we were going to do for this iteration of this issue. The [built in list ](https://github.com/dotnet/SqlClient/blob/main/src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Reliability/SqlConfigurableRetryFactory.cs)of current retriable errors doesn't include deadlocks and because of that we'll have to create our own. Of course we can do that but again, I thought we were keeping the code churn down for this initial pass.